### PR TITLE
Homework11: Aksana Batyrova

### PR DESCRIPTION
**Pull request by AksanaBatyrova**
HW 11-Decorator

### CODE REVIEW

File: homework_11.py
Comment (Stml89) [inline homework_11.py:1]: Please, use 1 file for 1 task

File: homework_11.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    return check_args
+
+
+@positive_arguments
+def example_function(*args):
+    """This function sum all arguments it takes"""
+    return sum(args)
+
+
+example_function(5, -4)
```
Comment (Stml89) [inline homework_11.py:38]:
```suggestion
assert example_function(5, -4) == <expected value>
```

File: homework_11.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+    function: function to check arguments of which
+
+    return: function that check arguments for type and positivity
+    """
+    def check_args(*args):
+        for arg in args:
+            if not isinstance(arg, (int, float)) or arg <= 0:
+                raise ValueError(f'{arg} is not a positive number')
+        return function
```
Comment (Stml89) [inline homework_11.py:28]: You are not calling `example_function` function, what you are doing here is just return function object
```suggestion
        return function(*args)
```

Comment (Stml89) [reply homework_11.py:28]: The same problem for task#2, task#3

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_11.py (hunk 1 above)
Comment (AksanaBatyrova) [reply homework_11.py:38]: This decorator throws exception if any of arguments passed to the function is negative, so my expected value is Value Error with some message in it. I haven't found the way how to assert exceptions, could you help, please?

File: homework_11.py (hunk 2 above)
Comment (AksanaBatyrova) [reply homework_11.py]: fixed

**Review by Stml89 — APPROVED** [review]
In general PR looks good, fix last comment and you can merge it

File: homework_11.py (hunk 1 above)
Comment (AksanaBatyrova) [reply homework_11.py]: fixed

**Pull request merged**